### PR TITLE
make store switcher style more resistent

### PR DESCRIPTION
**Pull request by Flyingmana**
make store switcher more resistent against moving of layout blocks
especially, moving the `navigation.sections` into the `header-wrapper`

### Description (*)
The issue I encountered was related to the duplicated store switcher block for the mobile navigation, which is part of the `navigation.sections`. 

The styling which is here added for the dropdown of the store switcher for the desktop variant, is currently also applied to the mobile variant when moved into the `header-wrapper` (to have the category navigation on Desktop between logo and other Header Parts)

### Fixed Issues (if relevant)

1. magento/magento2#<issue_number>: Issue title

### Manual testing scenarios (*)
1. have a layout xml which moves the `navigation.sections`into the `header-wrapper``
```
<page xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" layout="2columns-left" xsi:noNamespaceSchemaLocation="urn:magento:framework:View/Layout/etc/page_configuration.xsd" label="Customer My Account (All Pages)" design_abstraction="custom">
    <body>

        <move element="navigation.sections" destination="header-wrapper" after="logo"/>
    </body>
</page>
```
2. reduce the Website to "mobile size" and check the store switcher in the mobile navigation

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by ptylek** [comment]
@magento give me test instance

**Comment by magento-engcom-team** [comment]
Hi @ptylek. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @ptylek, here is your [new Magento instance](https://pr-26879.instances.magento-community.engineering).
Admin access: [https://pr-26879.instances.magento-community.engineering/admin_ae96](https://pr-26879.instances.magento-community.engineering/admin_ae96)
Login: `68996f57` Password: `ab5879cb3fc9`
Instance will be terminated in up to 3 hours.

File: app/design/frontend/Magento/luma/Magento_Theme/web/css/source/_module.less [hunk 1]
```diff
@@ -289,7 +289,7 @@
         text-align: center;
     }
 
-    .page-header,
+    .page-header .panel.wrapper,
```
Comment (ptylek) [inline app/design/frontend/Magento/luma/Magento_Theme/web/css/source/_module.less:292]: Hi @Flyingmana , thanks for contribution. Wouldn't it be enough to just limit the class to `.page-header .panel`?

**Review by ptylek — CHANGES_REQUESTED** [review]

File: app/design/frontend/Magento/luma/Magento_Theme/web/css/source/_module.less (hunk 1 above)
Comment (Flyingmana) [reply app/design/frontend/Magento/luma/Magento_Theme/web/css/source/_module.less:292]: probably not, as there is also .panel.header occurrence one level deeper in html.

image

**Review by ptylek — DISMISSED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ptylek, thank you for the review.
[ENGCOM-7061](https://jira.corp.magento.com/browse/ENGCOM-7061) has been created to process this Pull Request

**Comment by magento-engcom-team** [comment]
@Flyingmana thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Review by engcom-Alfa — CHANGES_REQUESTED** [review]
Hi @Flyingmana 

During testing, we faced the issue.

**_Problem:_** The switch sign is in the wrong position

**Manual testing scenario:**

1. have a layout XML which moves the navigation.sectionsinto the `header-wrapper`;

        <move element="navigation.sections" destination="header-wrapper" after="logo"/>
2. reduce the Website to "mobile size" and check the store switcher in the mobile navigation;

### Actual Result:
**Without fix:**   :x: 
after3

**With fix:** :heavy_check_mark: 
after1

3. Back to the desktop view and click on the switcher button

**Actual Result:**
:x: The switch sign has incorrect position
after2

@Flyingmana Could you take a look?

Thanks!

**Comment by ptylek** [comment]
Hi @Flyingmana, I close this PR due to inactivity. If you want to continue your work, simply reopen this PR. Thanks!

**Comment by vasilii-b** [comment]
@magento give me 2.4-develop instance

**Comment by magento-engcom-team** [comment]
Hi @vasilii-b. Thank you for your request. I'm working on Magento 2.4-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @vasilii-b, here is your [Magento instance](https://i-26879-2-4-develop.instances.magento-community.engineering).
Admin access: [https://i-26879-2-4-develop.instances.magento-community.engineering/admin_7dc8](https://i-26879-2-4-develop.instances.magento-community.engineering/admin_7dc8)
Login: `af567407` Password: `1a9d0b96a1a1`
Instance will be terminated in up to 3 hours.

**Pull request closed without merging**